core/server/utils/url.js

```diff
@@ -259,7 +259,7 @@ function urlFor(context, data, absolute) {
             urlPath = data.image;
             imagePathRe = new RegExp('^' + getSubdir() + '/' + STATIC_IMAGE_URL_PREFIX);
             absolute = imagePathRe.test(data.image) ? absolute : false;
-            secure = data.image.secure;
+            secure = data.image.secure || secure;
```

Comment:
Sorry for late review!

I just double checked this line of code. `data.image.secure` is never a case. We expect `image` to be the url path. If you remove the whole line, it should work as expected.
Thanks for adding the test 👍